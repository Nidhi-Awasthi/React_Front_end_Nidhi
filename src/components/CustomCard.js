import Button from "@material-ui/core/Button";
import Card from "@material-ui/core/Card";
import CardActionArea from "@material-ui/core/CardActionArea";
import CardActions from "@material-ui/core/CardActions";
import CardContent from "@material-ui/core/CardContent";
/***material ui components */
import { makeStyles } from "@material-ui/core/styles";
import Typography from "@material-ui/core/Typography";
import React from "react";
import CustomDrawer from "./Drawer";

const useStyles = makeStyles((theme) => ({
  root: {
    maxWidth: 295,
    [theme.breakpoints.down("xs")]: {
      maxWidth: "100%",
    },
  },
  title: {
    fontSize: 14,
  },
  pos: {
    marginBottom: 3,
    textAlign: "left",
    whiteSpace: "nowrap",
    textOverflow: "ellipsis",
    overflow: "hidden",
  },
  shortDesc: {
    textAlign: "left",
    whiteSpace: "nowrap",
    textOverflow: "ellipsis",
    overflow: "hidden",
  },
}));

const CustomCard = (props) => {
  const classes = useStyles();
  const { card } = props;
  const [drawerOpen, setDrawerOpen] = React.useState(false);
  const [cardDetail, setCardDetail] = React.useState(null);
  const toggleDrawer = (event) => {
    if (
      event &&
      event.type === "keydown" &&
      (event.key === "Tab" || event.key === "Shift")
    ) {
      return;
    }
    setDrawerOpen(!drawerOpen);
    setCardDetail(card);
  };

  return (
    <Card className={classes.root} onClick={(e) => toggleDrawer(e)}>
      <CardActionArea>
        <CardContent>
          <Typography
            className={classes.title}
            color="textSecondary"
            gutterBottom
          >
            {card.coreData.state}
          </Typography>
          <Typography gutterBottom variant="h5" component="h2">
            {card.coreData.number}
          </Typography>
          <Typography className={classes.pos} color="textSecondary">
            Application: {card.coreData.application}
          </Typography>
          <Typography className={classes.pos} color="textSecondary">
            Assignee: {card.coreData.assignee}
          </Typography>
          <Typography
            variant="body2"
            component="p"
            className={classes.shortDesc}
          >
            {card.coreData.shortDescription}
          </Typography>
        </CardContent>
      </CardActionArea>
      <CardActions>
        <Button size="small">Learn More</Button>
      </CardActions>
      <CustomDrawer
        toggleDrawer={toggleDrawer}
        drawerOpen={drawerOpen}
        cardDetail={cardDetail}
      />
    </Card>
  );
};
export default CustomCard;
