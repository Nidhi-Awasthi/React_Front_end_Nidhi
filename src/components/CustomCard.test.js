// // customCard.test.js
import React from "react";
import renderer from "react-test-renderer";
import CustomCard from "./CustomCard";

test("CustomCard", () => {
  const component = renderer.create(
    <CustomCard
      card={{
        coreData: {
          state: "New",
          number: "INC0010128",
          application: "application value",
          assignee: "System Administrator",
          shortDescription: "No approved expenses",
        },
      }}
    />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
