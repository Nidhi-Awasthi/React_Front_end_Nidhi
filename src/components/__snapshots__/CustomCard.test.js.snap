// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomCard 1`] = `
<div
  className="MuiPaper-root MuiCard-root makeStyles-root-1 MuiPaper-elevation1 MuiPaper-rounded"
  onClick={[Function]}
>
  <button
    className="MuiButtonBase-root MuiCardActionArea-root"
    disabled={false}
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <div
      className="MuiCardContent-root"
    >
      <p
        className="MuiTypography-root makeStyles-title-2 MuiTypography-body1 MuiTypography-colorTextSecondary MuiTypography-gutterBottom"
      >
        New
      </p>
      <h2
        className="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
      >
        INC0010128
      </h2>
      <p
        className="MuiTypography-root makeStyles-pos-3 MuiTypography-body1 MuiTypography-colorTextSecondary"
      >
        Application: 
        application value
      </p>
      <p
        className="MuiTypography-root makeStyles-pos-3 MuiTypography-body1 MuiTypography-colorTextSecondary"
      >
        Assignee: 
        System Administrator
      </p>
      <p
        className="MuiTypography-root makeStyles-shortDesc-4 MuiTypography-body2"
      >
        No approved expenses
      </p>
    </div>
    <span
      className="MuiCardActionArea-focusHighlight"
    />
  </button>
  <div
    className="MuiCardActions-root MuiCardActions-spacing"
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textSizeSmall MuiButton-sizeSmall"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        Learn More
      </span>
    </button>
  </div>
  <div />
</div>
`;
